### PR TITLE
Add end to end tests

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
- [x] Run docker compose in CI (new action, need to grab docker stuff from otp-middleware repo)
- [ ] clean up e2e file a little bit, extract some repeated code

blocked by #62

### CODE REVIEW

**Review by philip-cline — APPROVED** [review]
Just the relatively small stuff about environment variables when setting this up locally.

**Comment by philip-cline** [comment]
wow so good it gets a double approval from me I guess

File: package.json [hunk 1]
```diff
@@ -14,7 +14,8 @@
     "lint-code": "tsdx lint components pages util --quiet",
     "lint-docs": "documentation lint {components,pages,util}/**/*.js",
     "lint-types": "yarn tsc",
-    "predev": "yarn"
+    "predev": "yarn",
+    "test": "cp -r __tests__/e2e/env.test .env.build && yarn jest __tests__/e2e/"
```
Comment (binh-dam-ibigroup) [inline package.json:18]: Include e2e in the script name (and propagate to the yaml scripts)

File: __tests__/util/waitForDownload.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+          file.includes(downloadedFileName) && !file.includes('.crdownload')
+        )
+      })
+
+      if (downloadedFileFound) {
+        clearInterval(check)
+        resolve()
+      }
+
+      if (waitAttempts++ > 75) {
```
Comment (binh-dam-ibigroup) [inline __tests__/util/waitForDownload.ts:25]: Extract the limit to a constant.

**Review by binh-dam-ibigroup — APPROVED** [review]
LGTM, not much to complain about!

File: package.json (hunk 1 above)
Comment (miles-grant-ibigroup) [reply package.json]: Added in ee7126aa0a34c6d7926d6496ed753c56e3098e76!

**Pull request merged**